### PR TITLE
metadata gets carried when using methods

### PR DESCRIPTION
**Pull request by xHacka**

### CODE REVIEW

**Comment by xHacka** [comment]
`def from_cache_file(cls, filename) -> "Data":`
Method will not save metadata of previous object.
Reason: Its a static method, which means it doesnt have access to previous object properties.

**Comment by xHacka** [comment]
Should pages unittests be added to DS lib or LwDP?

File: th2_data_services/data.py [hunk 1]
```diff
@@ -693,3 +700,6 @@ def clear_cache(self):
         else:
             if self.__is_cache_file_exists():
                 self.__delete_cache()
+
+    def set_metadata(self, metadata: Dict):  # noqa
+        self.metadata = metadata
```
Comment (ConnectDIY) [inline th2_data_services/data.py:705]: It's common mistake to assign list or dict like this

When you do this, you assign a link to the object, but we need a deep copy.

Comment (ConnectDIY) [reply th2_data_services/data.py:705]: Let's also remake to `update_metadata` if you want to do it like a public method

**Review by ConnectDIY — CHANGES_REQUESTED** [review]

**Review by ConnectDIY — APPROVED** [review]

**Pull request merged**